elements/bulbs-video/elements/carousel.test.js

```diff
@@ -0,0 +1,190 @@
+      it('does not scroll to the video player', () => {
+        subject.attachedCallback();
+        expect(subject.scrollPlayerIntoView).not.to.have.been.called;
+      });
+    });
+  });
+
+  describe('detachedCallback', () => {
+    beforeEach(() => {
+      sinon.spy(subject.videoPlayer, 'removeEventListener');
+    });
+
+    it('removes playerEnded handler from videoPlayer', () => {
+      subject.detachedCallback();
+      expect(subject.videoPlayer.removeEventListener).to.have.been.calledWith(
+        'ended', subject.playerEnded
+      );
+    });
+  });
+
+  describe('scrollPlayerIntoView', () => {
+    // strangly, resetting scrollBy does not work
+    beforeEach(() => {
+      sinon.spy(window, 'scrollBy');
+    });
+
+    afterEach(() => {
+      window.scrollBy.restore();
+    });
+
+    it('calls videoplayer.scrollIntoView', () => {
+      sinon.spy(subject.videoPlayer, 'scrollIntoView');
+      subject.scrollPlayerIntoView();
+      expect(subject.videoPlayer.scrollIntoView).to.have.been.called;
+    });
+
+    context('there is a page <header>', () => {
+      let header;
+      beforeEach(() => {
+        header = document.createElement('header');
+        document.body.appendChild(header);
+        header.setAttribute('style', `
+          width: 100px;
+          height: 100px;
+        `);
+      });
+
+      afterEach(() => {
+        document.body.removeChild(header);
+      });
+
+      it('engages vertical scroll by the header height', () => {
+        subject.scrollPlayerIntoView();
+        expect(window.scrollBy).to.have.been.calledWith(0, 100);
+      });
+    });
+
+    context('there is no page <header>', () => {
+      it('does not engage any extra scrolling', () => {
+        subject.scrollPlayerIntoView();
+        expect(window.scrollBy).not.to.have.been.called;
+      });
+    });
+  });
+
+  describe('getActiveCarouselItem', () => {
+    context('an item is active', () => {
+      beforeEach(() => {
+        container.querySelector('#second').setAttribute(
+          'href', window.location.pathname
+        );
+      });
+
+      it('returns that item', () => {
+        expect(subject.getActiveCarouselItem()).to.eql(
+          container.querySelector('#second')
+        );
+      });
+    });
+
+    context('multiple items are active', () => {
+      beforeEach(() => {
+        container.querySelector('#first').setAttribute(
+          'href', window.location.pathname
+        );
+        container.querySelector('#second').setAttribute(
+          'href', window.location.pathname
+        );
+      });
+
+      it('returns the first active item', () => {
+        expect(subject.getActiveCarouselItem()).to.eql(
+          container.querySelector('#first')
+        );
+      });
+    });
+
+    context('no items are active', () => {
+      it('returns null', () => {
+        expect(subject.getActiveCarouselItem()).to.eql(null);
+      });
+    });
+  });
+
+  describe('playerEnded', () => {
```

Comment:
Yes, I'll go get these in. Foggy head. Need to add a snippet for `xdescribe`, `xcontext`, and `xit` so I don't miss these.